imagetest/test_suites/image_boot/image_boot_test.go

```diff
@@ -17,11 +18,84 @@ import (
 	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
 )
 
+// The values have been decided based on running spot tests for different images.
+var imageFamilyBootTimeThresholdMap = map[string]float64{
+	"centos":      60,
+	"debian":      50,
+	"rhel":        60,
+	"rocky-linux": 60,
+	"sles-12":     85,
+	"sles-15":     120,
+	"ubuntu":      75,
+	"ubuntu-pro":  110,
+}
+
 const (
 	markerFile     = "/boot-marker"
 	secureBootFile = "/sys/firmware/efi/efivars/SecureBoot-8be4df61-93ca-11d2-aa0d-00e098032b8c"
 )
 
+func getThresholdValue(image string) float64 {
+	names := make([]string, 0, len(imageFamilyBootTimeThresholdMap))
+	for name := range imageFamilyBootTimeThresholdMap {
+		names = append(names, name)
+	}
+	sort.Sort(sort.Reverse(sort.StringSlice(names)))
+	for _, name := range names {
+		if strings.Contains(image, name) {
+			return imageFamilyBootTimeThresholdMap[name]
+		}
+	}
+	return 0
+}
+
+func lookForProcesses() error {
```

Comment:
Dropping some ideas, why instead of looking for the process you don't look for the systemd service? that way you can detect if the service has failed or not - distinguishing a service failure from a time regression.

Imagine the case when the ssh service has failed to start (for any reason) and you're testing it, the process will never show up you will assume that ssh is taking too long but in fact the service is broken, if you test using the systemd service you will be aware of the service status and distinguish between time regression and the service failure, we'll at least have a better view of the real problem.

Reply:
i advised him to look at processes and not systemd service. we have a timeout on the check.

Reply:
How would the timeout impact using or not using/relying on systemd? One could even use systemd-analyze to determine every piece of the boot process time, then use blame to factor the time of the unit you desire, you can get much more accurate data from that.

Reply:
i'm sorry, i somehow resolved this by accident. i think using systemd-analyze as you suggest would be useful for per-unit determinations, but we only care about the final goal state, a single measurement. and we want to be reliable to errors or misconfigurations in the systemd space. so talking to systemd is not appropriate for this particular test